var/spack/repos/builtin/packages/relion/package.py

```diff
@@ -49,6 +58,11 @@ class Relion(CMakePackage):
     depends_on('fftw+float+double')
     depends_on('fltk', when='+gui')
     depends_on('cuda@8.0:8.99', when='+cuda')
+    # compiles with gcc > 5 under cuda8
+    conflicts('%gcc@5.0.0:9.0.0', when='+cuda')
+    # cuda 9 not yet supported
+    #  https://github.com/3dem/relion/issues/296
+    conflicts('cuda@9')
```

Comment:
Note that since you say `depends_on('cuda@8.0:8.99', when='+cuda')` above that this is not strictly required: if some other package attempts to request `cuda@9` then this conflict will be detected as part of concretization (that being said if this is removed it seems useful to keep the comment referring to the relion issue, but perhaps just place it next to the `depends_on` constraint).

Reply:
Ah yes.  There's no way to install cuda9 with the current ```depends_upon.```